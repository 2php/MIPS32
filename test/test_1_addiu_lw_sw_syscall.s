text:
  .text 0
  lw $2, addend_a($0) # r2 := addend_a
  addiu $3, $2, 456   # r3 := r2 + 456
  sw $3, result($0)   # result := r3
  lw $4, result($0)   # r4 := result
  syscall
addend_a:
  .word 123
result:
  .word 0
