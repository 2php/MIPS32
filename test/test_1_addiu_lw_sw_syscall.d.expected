booting
loading memory from file
00000000 3c020000
00000004 8c420020
00000008 244301c8
0000000c 3c010000
00000010 ac230024
00000014 3c040000
00000018 8c840024
0000001c 0000000c
00000020 0000007b
00000024 00000000

fetch instr @ 0x00000000 (00000000000000000000000000000000)
Mem[0x00000000] :  0x3c020000 (00111100000000100000000000000000)
001111 | 00000 | 00010 | 0000000000000000
LUI      rs: 0   rt: 2   imm:0x0000
R[ 2] <= 0x00000000 (00000000000000000000000000000000)

fetch instr @ 0x00000004 (00000000000000000000000000000100)
Mem[0x00000004] :  0x8c420020 (10001100010000100000000000100000)
100011 | 00010 | 00010 | 0000000000100000
LW       rs: 2   rt: 2   imm:0x0020
R[ 2] :  0x00000000 (00000000000000000000000000000000)
alu_a <= 0x00000000 (00000000000000000000000000000000)
alu_b <= 0x00000020 (00000000000000000000000000100000)
alu_r :  0x00000020 (00000000000000000000000000100000)
Mem[0x00000020] :  0x0000007b (00000000000000000000000001111011)
R[ 2] <= 0x0000007b (00000000000000000000000001111011)

fetch instr @ 0x00000008 (00000000000000000000000000001000)
Mem[0x00000008] :  0x244301c8 (00100100010000110000000111001000)
001001 | 00010 | 00011 | 0000000111001000
ADDIU    rs: 2   rt: 3   imm:0x01c8
R[ 2] :  0x0000007b (00000000000000000000000001111011)
alu_a <= 0x0000007b (00000000000000000000000001111011)
alu_b <= 0x000001c8 (00000000000000000000000111001000)
alu_r :  0x00000243 (00000000000000000000001001000011)
R[ 3] <= 0x00000243 (00000000000000000000001001000011)

fetch instr @ 0x0000000c (00000000000000000000000000001100)
Mem[0x0000000c] :  0x3c010000 (00111100000000010000000000000000)
001111 | 00000 | 00001 | 0000000000000000
LUI      rs: 0   rt: 1   imm:0x0000
R[ 1] <= 0x00000000 (00000000000000000000000000000000)

fetch instr @ 0x00000010 (00000000000000000000000000010000)
Mem[0x00000010] :  0xac230024 (10101100001000110000000000100100)
101011 | 00001 | 00011 | 0000000000100100
SW       rs: 1   rt: 3   imm:0x0024
R[ 1] :  0x00000000 (00000000000000000000000000000000)
R[ 3] :  0x00000243 (00000000000000000000001001000011)
alu_a <= 0x00000000 (00000000000000000000000000000000)
alu_b <= 0x00000024 (00000000000000000000000000100100)
alu_r :  0x00000024 (00000000000000000000000000100100)
Mem[0x00000024] <= 0x00000243 (00000000000000000000001001000011)

fetch instr @ 0x00000014 (00000000000000000000000000010100)
Mem[0x00000014] :  0x3c040000 (00111100000001000000000000000000)
001111 | 00000 | 00100 | 0000000000000000
LUI      rs: 0   rt: 4   imm:0x0000
R[ 4] <= 0x00000000 (00000000000000000000000000000000)

fetch instr @ 0x00000018 (00000000000000000000000000011000)
Mem[0x00000018] :  0x8c840024 (10001100100001000000000000100100)
100011 | 00100 | 00100 | 0000000000100100
LW       rs: 4   rt: 4   imm:0x0024
R[ 4] :  0x00000000 (00000000000000000000000000000000)
alu_a <= 0x00000000 (00000000000000000000000000000000)
alu_b <= 0x00000024 (00000000000000000000000000100100)
alu_r :  0x00000024 (00000000000000000000000000100100)
Mem[0x00000024] :  0x00000243 (00000000000000000000001001000011)
R[ 4] <= 0x00000243 (00000000000000000000001001000011)

fetch instr @ 0x0000001c (00000000000000000000000000011100)
Mem[0x0000001c] :  0x0000000c (00000000000000000000000000001100)
000000 | 00000 | 00000 | 00000 | 00000 | 001100
SPECIAL  rs: 0   rt: 0   rd: 0   sa: 0   SYSCALL
halt
